### DISCUSSION TITLE
Can no longer SFTP after upgrading Roborock S5 firmware

### DISCUSSION THREAD

**Discussion by JakeShirley** [Hypfer/Valetudo#1956] [Q&A / Support] [answered]
I started with the firmware version `1886` build, then upgraded to `2008` and then `2034`.

I can SSH with root and my private key, but when trying to SFTP (using FileZilla and Termius), I am getting a disconnect that looks like this (in FileZilla):
```
Command: 	open "root@10.1.0.199" 22
Trace:        	Looking up host "10.1.0.199" for SSH connection
Trace:        	Connecting to 10.1.0.199 port 22
Trace:        	We claim version: SSH-2.0-FileZilla_3.66.4
Trace:        	Connected to 10.1.0.199
Trace:        	Remote version: SSH-2.0-dropbear_2020.81
Trace:        	Using SSH protocol version 2
Trace:        	Doing ECDH key exchange with curve Curve25519 and hash SHA-256 (unaccelerated)
Trace:        	Server also has ecdsa-sha2-nistp256/ssh-dss/rsa-sha2-256/ssh-rsa host keys, but we don't know any of them
Trace:        	Host key fingerprint is:
Trace:        	ssh-ed25519 255 SHA256:lP/rWlaTw/rUa8L/ZePnsrV21+3cZVrHUzTGuadIrLk
Trace:        	Initialised AES-256 SDCTR (AES-NI accelerated) outbound encryption
Trace:        	Initialised HMAC-SHA-256 (unaccelerated) outbound MAC algorithm
Trace:        	Initialised AES-256 SDCTR (AES-NI accelerated) inbound encryption
Trace:        	Initialised HMAC-SHA-256 (unaccelerated) inbound MAC algorithm
Trace:        	Pageant is running. Requesting keys.
Trace:        	Pageant has 1 SSH-2 keys
Trace:        	Successfully loaded 1 key pair from file
Status:      	Using username "root". 
Trace:        	Trying Pageant key #0
Trace:        	Server refused our key
Trace:        	Offered public key from "/Users/jakeshirley/Downloads/j6595a773dc901-keys/j6595a773dc901.id_rsa"
Trace:        	Offer of public key accepted, trying to authenticate using it.
Trace:        	Sent public key signature
Trace:        	Access granted
Trace:        	Opening main session channel
Trace:        	Opened main channel
Trace:        	Started a shell/command
Status:      	Connected to 10.1.0.199
Error:         	FATAL ERROR: Received unexpected end-of-file from SFTP server
Trace:        	Got eof from child process
Trace:        	CControlSocket::DoClose(64)
Trace:        	CControlSocket::ResetOperation(66)
Trace:        	CSftpConnectOpData::Reset(66) in state 3
Error:         	Could not connect to server
Trace:        	CFileZillaEnginePrivate::ResetOperation(66)
```

Prior to upgrading I could SFTP with both programs.  I am a little out of my depth here :). It seems like the keys are accepted but the connection is cut off.

**Comment by Hypfer** [comment, marked as answer]
Starting with FW 2008, roborock switched the S5 to their then-new firmware structure they've introduced for the smaller flash on the S5 Max etc. These formerly-new firmwares aren't the regular ubuntu 14.04 anymore - as it was in older firmwares - but a heavily stripped down version of it to save space.

I'm frankly too lazy to check and validate this, but my assumption would be that 1886 still came with openSSHd and 2008 then switched to dropbear - the latter can be seen in your log. Dropbear simply doesn't implement sftp

Not sure if you can't use filezilla at all or if you can use the scp protocol - which is implemented by dropbear - there. Filezilla was never mentioned in any rooting guide so it's out of scope.
If you want to scp to a firmware with dropbear from a host featuring the [openSSH client version 9.0 and up](https://www.openssh.com/txt/release-9.0), you may have to add the `-O` flag as with that version, they've switched the scp command from using the scp protocol to by-default using the sftp protocol which isn't implemented by dropbear. `-O` tells it to use the old scp protocol again

**Reply by JakeShirley** [reply]
Great, thanks for the background @Hypfer!  I'll try out scp.

**Reply by JakeShirley** [reply]
Just wanted to hop back in and say that scp with `-O` on my M2 Macbook worked great, thanks again for the product and prompt support @Hypfer!
